tests/purs/passing/DerivingFunctorPrefersSimplerClasses.purs

```diff
@@ -0,0 +1,46 @@
+module Main where
+
+import Prelude
+
+import Data.Bifunctor (class Bifunctor)
+import Data.Profunctor (class Profunctor)
+import Effect (Effect)
+import Effect.Console (log)
+import Test.Assert (assert')
+
+newtype MonoAndBi a b = MonoAndBi (Effect Unit)
+derive instance Functor (MonoAndBi a)
+instance Bifunctor MonoAndBi where
+  bimap _ _ _ = MonoAndBi (assert' "Bifunctor instance was used but the Functor instance was expected" false)
+
+newtype Test1 a = Test1 (MonoAndBi Int a)
+derive instance Functor Test1
+
+data ExclusivelyBi a b
+derive instance Bifunctor ExclusivelyBi
```

Comment:
This surprised me. How does a non-inhabited type have a Bifunctor (or later Profunctor) instance?

Reply:
Same way it has a Functor instance—trivially. `map _ = absurd`, `bimap _ _ = absurd`, `dimap _ _ = absurd`, etc.

Except not literally `absurd`, because uninhabited types aren't literally `Void`. If #2922 were implemented I'd say it's equivalent to that, but instead I'm afraid we just have to accept that the compiler can implement these functions where a user would have to resort to the sort of (mild!) hack used in [`Data.Void`](https://github.com/purescript/purescript-prelude/blob/v6.0.1/src/Data/Void.purs#L31-L31).